### ISSUE TITLE
Problems when converting from PC to Mac

### ISSUE THREAD

**Issue by vberry** [rscustom/rocksmith-custom-song-toolkit#207]
Hi developpers, 
on one song over 5 or 6 there is a problem in converting an RS song (CDLC) from PC to Mac. 
Usually the application gets stuck (for instance with the song U2_One_Live_v1_p.psarc), but this time I got an error message asking to send you the log (which I found). This was on the french song 00 - Louise_Attaque_Jtemmène_au_vent_v1.3_p.psarc, see below the log. 
Thanks

RocksmithToolkitLib
Void InflateEntry(RocksmithToolkitLib.PSARC.Entry, RocksmithToolkitLib.PSARC.BigEndianBinaryReader, UInt32[], UInt32, System.IO.Stream)
Exception caught:

 *removed change log, due it's useless*

**Comment by L0FKA** [comment]
Sorry for long delay, but it seems that you have outdated Toolkit, we've fixed this kind of error in past days, sorry that autoupdater is disabled for MACs, you can run it manually or install a fresh copy of the toolkit from our main page ;)

**Comment by vberry** [comment]
No, I think it's not the problem: I've updated with a quite recent version already: I'm running V 2.6.1.0-84a042f6 (extract from the "about" box in the tool).

**Comment by L0FKA** [comment]
All should be fine, I'll test with Linux, surely all'll be fine too, but have you this for all the psarc's? or just for thees one?

**Comment by vberry** [comment]
Ok, I found a newer version of RSInjector.app on http://www.rscustom.net (854ko dating from May, 29, 2014):
- now the Creep-Radiohead works (it didn't before) : yeah !!!!
- but the Niagara song "J'ai vu" doesn't :(
- For the U2 One Live, I'm still stuck at the conversion step.....

Thanks for testing under linux and telling me if you have issues also there.

**Comment by hajdbo** [comment]
I found the problem with Niagara song J'ai vu... It's a problem with the converter which uses FixShowlights() which breaks the working showlights and makes the song unplayable on Mac.
Manual fix: Unpack the PC package and unpack the Mac package; then overwrite the Mac showlights file  with the working PC showlights file. Then repack the Mac package.
I will open a new issue about this problem.

**Comment by hajdbo** [comment]
I converted U2 One live, Niagara J'ai Vu, Louise Attaque Viens Jtemmene au vent with the most recent build.
All converted with no toolkit errror.
U2 and Louise Attaque played fine in Rocksmith 2014.

However, "Niagara J'ai Vu" doesn't display the note highway and the game hangs.
I remember having a playable version a few days ago; I'll try and see the differences.
Since this is not a showlights problem anymore, I guess it might be related to the single quote.

**Comment by vberry** [comment]
> I converted U2 One live, Niagara J'ai Vu, Louise Attaque Viens 
> Jtemmene au vent with the most recent build.
Did you post new versions for these songs on custom forge? That would be great if the problem is fixed!

Niagara is really a fine song, hope you'll be able to fix this single quote problem.

Thanks for your help in this.

**Comment by hajdbo** [comment]
Very interesting, we have a regression!
On Mac, build 120 (https://ci.appveyor.com/project/rscustom/rocksmith-custom-song-toolkit/build/120/artifacts ) builds a working Niagara-JaiVu when following these steps:

1- convert PC->Mac
2- unpack Mac & unpack PC
3- copy the showlights.xml from PC to Mac
4- pack the Mac directory
--> working CDLC

When using the same steps (or only converting, as the newest build does not modify showlights.xml) with the latest build, game hangs.

**Comment by vberry** [comment]
Ok,
 after receiving you post I tested in parallel this afternoon:
- the latest release (gives me "2.6.0.1.84...." in About menu)
- build 120 for which you give a link ("2.6.10.3d...." in About menu). 
Doing the following for each tune I tested: 
1 - conversion from PC
2 - unpack PC and Mac version 
3 - take showlight file from PC to replace that of the MAC unpacked version
4 - repack MAC file
5 - test MAC file with RS (injector).

Here are what happened:
* Niagara works ok with build120, while RS hangs after tuning the guitar on version generated by latest build
* Louise Attaque: same story as above
* Jeff Buckley - Hallelujah (didn't I tell you I could not convert it neither ;) : I just had time to repeat the winning recipe for this one, then put it with my ok songs, but RS does not see it :(

So:
- thanks for giving the recipe (I can now play two more tunes, yeah !)
- the recipe is not universal (poor Jeff)
- thanks for fixing this showlight problem in a further release of the toolkit, as this is a pain to do (the GUI is not 100% OSX compatible and hangs sometimes ;), but good job enough to make my way through it.

Thanks guys

**Comment by hajdbo** [comment]
Let's try all builds between 120 and 141, to find the problematic commit.

**Comment by L0FKA** [comment]
just let me know If it showlight issue of is it something sng related or..? that would be faster

**Comment by hajdbo** [comment]
I don't know. Build 130 works like 120. I'll try 135 next.

**Comment by hajdbo** [comment]
build 135 works also.

**Comment by hajdbo** [comment]
Build 139 works also

**Comment by hajdbo** [comment]
Damned. My patch to keep the showlights.xml intact breaks something else.
To build a working Niagara-JaiVu, it seems we need to "first fix/break" the showlights.xml, then build the SNG and then put the original showlights.xml.

**Comment by cozy1** [comment]
@hajdbo Can we close this?

**Comment by L0FKA** [comment]
I've fixed CDT showlights bug, so It's probably done for now..

**Comment by vberry** [comment]
Thanks LOFKA for your modification. 
I will download the new version ASAP (next week maybe), and will give feedback. 


vincent

**Comment by hajdbo** [comment]
I confirm the showlights bug has been fixed. CDLC "Niagara - J'ai vu" and "Rod Stewart - Young Turks" were good tests; both works after converting with build 183. Great job guys.

**Comment by vberry** [comment]
Good to know about Niagara. 
I just tested Creedence Clearwater - I put a spell on you which wasn't rightly converted with the previous version, no it's ok. Thanks a lot !
